core/app/models/spree/shipment.rb

```diff
@@ -304,19 +279,6 @@ def update_attributes_and_order(params = {})
       end
     end
 
-    # Updates various aspects of the Shipment while bypassing any callbacks.  Note that this method takes an explicit reference to the
-    # Order object.  This is necessary because the association actually has a stale (and unsaved) copy of the Order and so it will not
-    # yield the correct results.
-    def update!(order)
-      old_state = state
-      new_state = determine_state(order)
-      update_columns(
-        state: new_state,
-        updated_at: Time.now,
-      )
-      after_ship if new_state == 'shipped' and old_state != 'shipped'
```

Comment:
this is concerning to me too. not only do we call `#after_ship` in an after transition so that we send out an email, there is the chance that we could call it again here. overall this method is terrifying as it alters the state of a shipment outside of the state machine.